src/azure-cli/azure/cli/command_modules/storage/operations/account.py

```diff
@@ -217,6 +219,21 @@ def create_storage_account(cmd, resource_group_name, account_name, sku=None, loc
     if enable_nfs_v3 is not None:
         params.enable_nfs_v3 = enable_nfs_v3
 
+    if enable_alw is not None:
+        ImmutableStorageAccount = cmd.get_models('ImmutableStorageAccount')
+        AccountImmutabilityPolicyProperties = cmd.get_models('AccountImmutabilityPolicyProperties')
+        immutability_policy = None
+
+        if immutability_period_since_creation_in_days is not None:
```

Comment:
What if `immutability_period_since_creation_in_days` is `None` but `immutability_policy_state` or `allow_protected_append_writes` is not?

Reply:
Right, I had the logic that all three are needed, let me change it.